File: src/Ferry.cs
using System.Linq;
using System.Collections.Generic;

namespace src{
    public class Ferry: Vehicle {
        private List<Destination> _portsVisited = new List<Destination>();

        public bool GoTo(Destination destination){
            if((new[] {
                Destination.MainWharf,
                Destination.PortLevy,
                Destination.QuailIsland})
                .Contains(destination)){

                _portsVisited.Add(destination);
                return true;
            }
            return false;
        }

        public int PortsVisitedCount{
            get{
                return _portsVisited.Count();
            }
        }
    }
}

File: py/src/Ferry.py
from .Vehicle import Vehicle
from .Destination import Destination


class Ferry(Vehicle):
    def __init__(self):
        self.__portCount = 0

    def go_to(self, destination: Destination) -> bool:
        if destination in [Destination.MainWharf, Destination.PortLevy, Destination.QuailIsland]:
            self.__portCount = self.__portCount + 1
            return True
        return False

    def ports_visited_count(self): return self.__portCount
